src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -245,7 +261,7 @@ private function parseDefinition($id, $service, $file, array $defaults)
             return;
         }
 
-        if (is_array($service) && array_values($service) === $service) {
+        if (is_array($service) && $this->isUsingShortSyntax($service)) {
```

Comment:
note that array_values checked the order of keys, whereas isUsingShortSyntax doesn't anymore, isn't it?